programs/server/config.xml

```diff
@@ -272,16 +271,67 @@
                 server - one of LDAP server names defined in 'ldap_servers' config section above.
                         This parameter is mandatory and cannot be empty.
                 roles - section with a list of locally defined roles that will be assigned to each user retrieved from the LDAP server.
-                        If no roles are specified, user will not be able to perform any actions after authentication.
-                        If any of the listed roles is not defined locally at the time of authentication, the authenthication attempt
-                         will fail as if the provided password was incorrect.
+                        If no roles are specified here or assigned during role mapping (below), user will not be able to perform any
+                         actions after authentication.
+                role_mapping - section with LDAP search parameters and mapping rules.
+                        The list of strings (values of attributes) returned by the search will be transformed and the resulting strings
+                         will be treated as local role names and assigned to the user.
+                        There can be multiple 'role_mapping' sections defined inside the same 'ldap' section. All of them will be
+                         applied.
+                    base_dn - template used to construct the base DN for the LDAP search.
+                            The resulting DN will be constructed by replacing all '{username}' and '{bind_dn}' substrings
+                             of the template with the actual user name and bind DN during each LDAP search.
```

Comment:
Where can `base_dn` be used? If the only place is `search_filter` then it seems we don't need `base_dn` cause everything can be put directly to `search_filter`.

Reply:
These is very LDAP-specific params, the base DN is specified in ldap_search routine as one of the parameters, the filter is another one. The base DN is a path where the search should start from. Changing their places or names would be very misleading for LDAP admins.

Reply:
As a substitution, `base_dn` may be used in the filter though. But that depend on the query.